app/services/reference_status.rb

```diff
@@ -40,6 +36,10 @@ def feedback_completed?
     application_references.select(&:feedback_provided?).size == 2
   end
 
+  def non_overdue_references
+    application_references.reject(&:feedback_overdue?)
+  end
```

Comment:
Do we need to `reject(&:feedback_overdue?)`?

Following from that, if the only place we use this is with the `select` from above, we could also hoist the `select` here and then rename this:

```suggestion
  def references_that_do_not_need_replacing
    application_references.select { |reference| reference.feedback_provided? || reference.feedback_requested? }
  end
```

Reply:
Could this also just be `application_references.reject(&:replacement?)`?

Reply:
I might be missing something but as feedback_overdue? is a method, not an enum value/attribute/boolean,  it means a reference can be both overdue and requested. Do we not need to reject them before checking how many are in a provided/requested state. (i.e. the ones that don't need replacing)?